arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,303 @@
+        optionsViewController.title = "Status"
+        show(optionsViewController, sender: self)
+    }
+    
+    /// Use the contingent values definition to generate the possible values for the protection field.
+    func showProtectionOptions() {
+        // Get the previously selected protection value.
+        let previouslySelectedProtection = selectedProtection
+        guard let feature = feature else { return }
+        // Get the contingent value results with the feature for the protection field.
+        let contingentValuesResult = featureTable?.contingentValues(with: feature, field: "Protection")
+        // Get contingent coded values by field group.
+        guard let protectionGroupContingentValues = contingentValuesResult?.contingentValuesByFieldGroup["ProtectionFieldGroup"] as? [AGSContingentCodedValue] else { return }
+        // Get the selected index if applicable.
+        let selectedIndex = protectionGroupContingentValues.firstIndex { $0.codedValue.name == selectedProtection?.codedValue.name }
+        // Use the coded value names as labels to show the option table view controller.
+        let optionsViewController = OptionsTableViewController(labels: protectionGroupContingentValues.map { $0.codedValue.name }, selectedIndex: selectedIndex) { [weak self] newIndex in
+            guard let self = self else { return }
+            // Set the selected protection value.
+            self.selectedProtection = protectionGroupContingentValues[newIndex]
+            feature.attributes["Protection"] = self.selectedProtection?.codedValue.code
+            // Set the attributes to nil if the protection value has changed.
+            if self.selectedProtection != previouslySelectedProtection, self.selectedBufferSize != nil {
+                self.selectedBufferSize = nil
+            }
+            // Remove the options view controller after selection.
+            self.navigationController?.popViewController(animated: true)
+        }
+        // Set the options view controller's title and present it.
+        optionsViewController.title = "Protection"
+        show(optionsViewController, sender: self)
+    }
+    
+    /// Get the minimum and maximum values of the possible buffer sizes.
+    func showBufferSizeOptions() {
+        guard let feature = feature else { return }
+        // Get the contingent value results using the feature and field.
+        let contingentValueResult = featureTable?.contingentValues(with: feature, field: "BufferSize")
+        guard let bufferSizeGroupContingentValues = contingentValueResult?.contingentValuesByFieldGroup["BufferSizeFieldGroup"] as? [AGSContingentRangeValue] else { return }
+        // Set the minimum and maximum possible buffer sizes.
+        let minValue = bufferSizeGroupContingentValues[0].minValue as! Int
+        let maxValue = bufferSizeGroupContingentValues[0].maxValue as! Int
+        bufferSizes = Array(minValue...maxValue)
```

Comment:
You might consider changing the type of `bufferSizes` to `ClosedRange<Int>`.

Reply:
I was also thinking about making this suggestion, but wasn't sure if a `ClosedRange` can be the data source of a picker.

Reply:
I don't think I'm able to since I can't return the value using an index from `ClosedRange`